### ISSUE TITLE
[Phase 5] Experiment Pipeline and Augmentation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/gravitational-wave-detection#6]
## Objective

Create end-to-end experiment pipeline with data augmentation and reproducible training.

**Part of:** #1

## Dependencies

- All previous phases (#2-#5) complete

### Experiment Runner
- [ ] Create `src/pipeline/runner.py` with `ExperimentRunner`
- [ ] Wire together data, features, model, evaluation
- [ ] Configuration via YAML

### Data Augmentation
- [ ] Create `src/pipeline/augmentation.py`
- [ ] Time shift (circular)
- [ ] Amplitude scaling
- [ ] Gaussian noise injection

### CLI Entry Point
- [ ] Create `scripts/run_experiment.py`
- [ ] Support model type, epochs, batch size arguments

### Experiment Runner

```python
# src/pipeline/runner.py
import yaml
from src.data.loader import GWDataLoader
from src.models.cnn1d import GWDetector1D
from src.models.trainer import ModelTrainer
from src.evaluation.metrics import DetectionMetrics

class ExperimentRunner:
    def __init__(self, config_path=None, **kwargs):
        if config_path:
            with open(config_path) as f:
                self.config = yaml.safe_load(f)
        else:
            self.config = kwargs

    def run(self):
        # Load data
        loader = GWDataLoader(
            self.config.get('dataset', 'g2net'),
            sample_size=self.config.get('sample_size', 10000)
        )
        X_train, X_test, y_train, y_test = loader.load_split()

        # Create dataloaders
        train_loader = self._create_dataloader(X_train, y_train, shuffle=True)
        test_loader = self._create_dataloader(X_test, y_test, shuffle=False)

        # Train
        model = GWDetector1D()
        trainer = ModelTrainer(model)

        for epoch in range(self.config.get('epochs', 50)):
            loss = trainer.train_epoch(train_loader)
            print(f"Epoch {epoch}: Loss = {loss:.4f}")

        # Evaluate
        y_proba = trainer.predict(test_loader)
        metrics = DetectionMetrics(y_test, y_proba)

        return {
            'auc': metrics.auc(),
            'efficiency_at_1e-3_far': metrics.efficiency_at_far(1e-3)[0]
        }
```

### Data Augmentation

```python
# src/pipeline/augmentation.py
import numpy as np

class GWAugmenter:
    def __init__(self, time_shift=True, amplitude_scale=True, noise_inject=True):
        self.time_shift = time_shift
        self.amplitude_scale = amplitude_scale
        self.noise_inject = noise_inject

    def augment(self, X, y):
        """Apply augmentations to batch."""
        X_aug = X.copy()

        if self.time_shift:
            # Circular shift by random amount
            shifts = np.random.randint(-512, 512, size=len(X))
            for i, shift in enumerate(shifts):
                X_aug[i] = np.roll(X_aug[i], shift, axis=-1)

        if self.amplitude_scale:
            # Scale by factor between 0.8 and 1.2
            scales = np.random.uniform(0.8, 1.2, size=(len(X), 1, 1))
            X_aug = X_aug * scales

        if self.noise_inject:
            # Add small Gaussian noise
            noise = np.random.normal(0, 0.01, X_aug.shape)
            X_aug = X_aug + noise

        return X_aug, y
```

## Definition of Done

- [ ] Full pipeline runs: `python scripts/run_experiment.py --epochs 50`
- [ ] Results logged to JSON
- [ ] Augmentation improves validation AUC
- [ ] Reproducible with random seed

## Technical Notes

- Time shift is safe due to signal location uncertainty in real data
- Amplitude scaling helps with varying SNR
- Be careful not to augment test set!
